etc/vcl_snippets/fetch.vcl

```diff
@@ -106,16 +106,30 @@
     if (beresp.ttl > 0s && (req.request == "GET" || req.request == "HEAD") && !req.http.x-pass ) {
         unset beresp.http.set-cookie;
 
-        # init surrogate keys
-        if (beresp.http.X-Magento-Tags) {
-            set beresp.http.Surrogate-Key = beresp.http.X-Magento-Tags " text";
-        } else {
-            set beresp.http.Surrogate-Key = "text";
-        }
+        # init surrogate keys if on outermost node
+        if (fastly.ff.visits_this_service == 0) {
```

Comment:
```suggestion
        if (fastly.ff.visits_this_service == 0 && req.restarts == 0) {
```

Got another tip from one of our internal teams... in those cases where a request undergoes a VCL 'restart', we want to avoid adding the keys again there too.